### DISCUSSION TITLE
INAV not discovering mq8 can-bus GPS on H743 fc

### DISCUSSION THREAD

**Discussion by zebediz** [iNavFlight/inav#7275] [Q&A]
Hi - I am new to INAV (hate the ardupilot interface and looking for something that works more intuitively). 

I have a Matek H743 flight controller and the Matek CAN bus m8q GPS.

I am using inav 3.0.1

After connecting it all and connecting to battery and computer, no compass or gps is discovered - just red icons and a loud beeping (not stopping). The gps was one thing that worked on Ardupilot so connection should be ok (flashing blue and green).

I tried the compass calibration but that didn't help. There seems to be some kind of packet error.

Here is a picture of my calibration screen on the iNav configurator:

![image](https://user-images.githubusercontent.com/44823656/125884987-f49449eb-06d4-4181-baa2-ffd0f8d96d81.png)

Any help much appreciated.

**Comment by JulioCesarMatias** [comment]
iNav still does not work with H7 + CAN

Use RX and TX from GPS and SCL and SDA from Compass

https://github.com/iNavFlight/inav/issues/7166

**Reply by zebediz** [reply]
ok - so I have to clip the wires and solder onto the board?

**Reply by JulioCesarMatias** [reply]
> ok - so I have to clip the wires and solder onto the board?

yes
